src/pg_cron.c

```diff
@@ -1671,22 +1671,36 @@ ManageCronTask(CronTask *task, TimestampTz currentTime)
 			#else
 				mq = shm_toc_lookup(toc, PG_CRON_KEY_QUEUE, false);
 			#endif
-			responseq = shm_mq_attach(mq, task->seg, NULL);
+
+			/*
+			 * We pass the DSM segment as NULL to avoid registering a callback,
+			 * since we want to be able to call shm_mq_attach multiple times.
+			 */
+			responseq = shm_mq_attach(mq, NULL, NULL);
```

Comment:
calling shm_mq_attach seems to be fine, since the worst damage it can do is a palloc